src/app/parcels/ActionBar/DuplicateDownloadWarning.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { ParcelsTableRow } from "@/app/parcels/getParcelsTableData";
+import { getParcelsWithEvent } from "@/app/parcels/fetchParcelTableData";
+import Alert from "@mui/material/Alert/Alert";
+import styled from "styled-components";
+import { StatusType } from "@/app/parcels/ActionBar/Statuses";
+
+export interface DuplicateDownloadWarningProps {
+    parcels: ParcelsTableRow[];
+    targetEventName: StatusType;
+}
+
+const StyledAlert = styled(Alert)`
+    border-radius: 4px;
+`;
+
+const DuplicateDownloadWarning: React.FC<DuplicateDownloadWarningProps> = async (props) => {
+    const parcelIds = props.parcels.map((parcel) => parcel.parcelId);
+    const { parcels, error } = await getParcelsWithEvent(props.targetEventName, parcelIds);
+    if (error) {
+        return;
+    }
+
+    const postcodes = parcels.map((parcel) => parcel.client_address_postcode);
+    const uniquePostcodes = Array.from(new Set(postcodes));
```

Comment:
I feel like this task of finding whether there'd be duplicate events should be delegated to the parent element. It feels a little confusing to see `<DuplicateDownloadWarning ...>` in the parent when the warning doesn't always exist. What do you think?

Reply:
I agree it could be confusing. My thinking was to have all the functionality in one component which can then be shared by both the shipping label and shopping list modals so that we could avoid repeating code/logic between the two. So I guess that's the tradeoff - do you have a preference? Or is there another way to do it?

Reply:
I'm thinking that the postcode retrieval should be a helper function (probably near `getParcelsWithEvent`?) so then you can reuse it for both, and then this component would only contain bits for the styled alert i.e. 
```
const StyledAlert = styled(Alert)`
    border-radius: 4px;
`;

return (
            <StyledAlert severity="warning">
                The following postcodes have already been downloaded: {uniquePostcodes.join(", ")}.
                Are you sure you want to print again?
            </StyledAlert>
        )
    );
```
and then add the component with `duplicateDownloadedPostcodes.lenght > 0 && </DuplicateDownloadWarning>` so it's clear it only appears when we have duplicate downloads. What do you think?

Reply:
Ok I've tried this out, lmk what you think!

Reply:
Left a comment on review :)